### DISCUSSION TITLE
WP headless with Elementor as builder

### DISCUSSION THREAD

**Discussion by Stelikas** [elementor/elementor#27304] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

Be able to use elementor as a headless builder through wordpress in other website platforms. 

### Describe the solution you'd like.

These past couple of days I've been trying to create a headless WP with elementor as a page builder for a website, although everything works fine, some widget have script dependencies which make it really hard to work with, at least, in my little experience on this, i have not found any methods which pull all the script dependencies through the API. 

So to sum things up, it would be nice to be able to request through the WP API all the data needed to display an elementor page, 
that means: 
1. HTML Content
2. Styles 
3. Javascript dependencies

### Describe alternatives you've considered.

We've been able to create an alternative API request which returns the internal styles of an elementor page like below: 

```
add_action("rest_api_init", function () {

    register_rest_route(
          "test/v2"
        , "/pages/(?P<id>\d+)/contentElementor"
        , [
            "methods" => "GET",
            "callback" => function (\WP_REST_Request $req) {

                $contentElementor = "";

                if (class_exists("\\Elementor\\Plugin")) {
                    $post_ID = $req->get_param("id");

                         $pluginElementor = \Elementor\Plugin::instance();
$contentElementor = $pluginElementor->frontend->get_builder_content_for_display($post_ID, $with_css = true);
                }


                return $contentElementor;

            },
        ]
    );


});
```

The only issue here is that some widgets have script dependencies and they do not work as intended, for example, the tabs widget or the toggles widget, do not work, they are not clickable. 

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by NearlyShake** [comment]
Would also be interested in this feature for sure for modern web development.
